Library/Homebrew/cmd/install.rb

```diff
@@ -298,6 +298,32 @@ def run
         Install.perform_preinstall_checks_once
         Install.check_cc_argv(args.cc)
 
+        overwrite = if GitHub::Actions.env_set? &&
+                       ENV["HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE"].blank? &&
+                       ENV["HOMEBREW_GITHUB_ACTIONS"].blank?
+          if ENV["HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE_WARNING"].blank?
+            message = <<~WARNING
+              In GitHub Actions, `brew install` behaves the same as `brew install --overwrite`.
+
+              To disable this behaviour, set `HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE`.
+
+              To silence this warning, set `HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE_WARNING`.
+            WARNING
+
+            puts GitHub::Actions::Annotation.new(:warning, message)
+
+            # Print warning only once.
+            github_env = ENV.fetch("GITHUB_ENV", "")
+            if File.exist?(github_env) && File.writable?(github_env)
+              File.open(github_env, "a") { |f| f.puts("HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE_WARNING=1") }
+            end
```

Comment:
I'm 50-50 about this bit, and don't mind dropping it.

Reply:
Also don't feel strongly either way. If it's Python only: can drop.